fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryManager.java

```diff
@@ -122,51 +127,52 @@ public CompletableFuture<Void> mergeIndex(@Nonnull LucenePartitioner partitioner
                 null,
                 ScanProperties.FORWARD_SCAN);
 
-        // TO DO: https://github.com/FoundationDB/fdb-record-layer/issues/2414
         return cursor
                 .map(tuple ->
                         Tuple.fromItems(tuple.getItems().subList(0, groupingCount)))
                 .forEachAsync(groupingKey -> mergeIndex(analyzerWrapper, groupingKey, partitioner, agilityContext),
-                        2);
+                        2)
+                .whenComplete((vignore, eignore) -> agilityContext.flushAndClose());
```

Comment:
I don't think we want to call `flushAndClose()` here if there is an error, only if it is successful. If there is an error, I think we want `abortAndReset`

Reply:
✅